src/transformers/utils/quantizers.py

```diff
@@ -0,0 +1,787 @@
+
+        elif self.quantization_method == QuantizationMethod.GPTQ:
+            quantizer_class = GPTQHFQuantizer
+            config_class = GPTQConfig
+
+        else:
+            # unknown quantization_method, which is not None:
+            raise NotImplementedError(
+                f"Unsupported quantization method detected: {self.quantization_method}. Check for updates."
+            )
+
+        # finishing touches ========================================================
+        if not isinstance(self.quantization_config, (dict, QuantizationConfigMixin)):
+            raise ValueError(
+                f"Invalid type for `quantization_config`: {type(self.quantization_config)}. Should be a `dict` or a"
+                " `QuantizationConfigMixin` subclass instance."
+            )
+
+        self.quantization_config = config_class.from_dict(
+            config_dict=dict(self.quantization_config)
+        )  # dict to class instance
+
+        config.quantization_config = self.quantization_config
+        quantizer = quantizer_class(
+            quantization_config=self.quantization_config, quantization_status=quantization_status
+        )
+        return quantizer
+
+
+def get_module_from_name(module, tensor_name: str) -> Tuple[Any, str]:
+    if "." in tensor_name:
+        splits = tensor_name.split(".")
+        for split in splits[:-1]:
+            new_module = getattr(module, split)
+            if new_module is None:
+                raise ValueError(f"{module} has no attribute {split}.")
+            module = new_module
+        tensor_name = splits[-1]
+    return module, tensor_name
+
+
+class HFQuantizer(ABC):
+    requires_parameters_quantization = False
+    aux_keys_suffixes = ()
+
+    @property
+    @abstractmethod
+    def method(self) -> QuantizationMethod:
+        """placeholder to method from QuantizationMethod"""
+        ...
+
+    def __init__(self, quantization_config: QuantizationConfig, quantization_status: QuantizationStatus, **kwargs):
+        self.quantization_status = quantization_status
+        self.modules_to_not_convert = []
+
+    def validate_environment(self, *args, **kwargs):
+        """checking env after instantiation, including modules presence and versions"""
+        ...
+
+    def process_model_before_weight_loading(self, model: PreTrainedModel, **kwargs) -> PreTrainedModel:
+        """setting model attributes and/or converting model BEFORE weights loading"""
+        model.is_quantized = True
+        model.quantization_method = self.method
+        model.is_loaded_in_4bit = False
+        model.is_loaded_in_8bit = False
+        logger.info("Activating quantized loading for this model using {self.__class__}")
+        return model
+
+    def process_model_after_weight_loading(self, model: PreTrainedModel, **kwargs):
+        """setting model attributes and/or converting model AFTER weights loading"""
+        model.eval()
```

Comment:
I would remove this and make it abstract as well, `model.eval()` is always call after `from_pretrained`
```suggestion
    @abstractmethod
    def process_model_after_weight_loading(self, model: PreTrainedModel, **kwargs):
        ...
```